### PR TITLE
Create Mutation Benchmarks

### PR DESCRIPTION
**Pull request by Hedgemon4**
Description
- Create runs to benchmark mutations

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -3,4 +3,4 @@
     url = https://github.com/Teamable-Analytics/algorithms-simulation-cache.git
 [submodule "api/ai/external_algorithms/group_matcher_algorithm/group-matcher"]
 	path = api/ai/external_algorithms/group_matcher_algorithm/group-matcher
-	url = git@github.com:ketphan02/group-matcher.git
+	url = https://github.com/ketphan02/group-matcher.git
```
Comment (justino599) [inline .gitmodules:6]: Changed this URL to make it consistent with the other submodule

File: benchmarking/data/simulated_data/realistic_class/providers.py [hunk 2]
```diff
@@ -132,6 +162,29 @@ def get(self, seed: int = None) -> List[Student]:
         return [students[_] for _ in order]
 
 
+class RealisticMockInitialTeamsProvider(InitialTeamsProvider):
+    def __init__(self, num_teams: int):
+        self.num_teams = num_teams
+
+    def get(self) -> List[TeamShell]:
+        projects = []
+        project_cycler = itertools.cycle(get_realistic_projects())
```
Comment (Opeyem1a) [inline benchmarking/data/simulated_data/realistic_class/providers.py:171]: nice

File: benchmarking/runs/mutations/mutation_benchmarking.py [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+            ],
+            "mutate_greedy_local_max_n_4": [
+                GreedyLocalMaxMutation(number_of_teams=4, num_mutations=max_spread)
+            ],
+            "mutate_greedy_local_max_n_8": [
+                GreedyLocalMaxMutation(number_of_teams=8, num_mutations=max_spread)
+            ],
+            "mutate_greedy_local_max_with_random_swap": [
+                GreedyLocalMaxMutation(num_mutations=(max_spread // 2)),
+                RandomSwapMutation(num_mutations=(max_spread // 2)),
```
Comment (Opeyem1a) [inline benchmarking/runs/mutations/mutation_benchmarking.py:101]: we're floor dividing a lot here, this gets funky if someone uses this code and messes with `max_spread` (i.e. making it an odd number), not really sure what the solution is though, maybe a comment? Maybe docs? This comment is just to illuminate the issue

File: benchmarking/runs/mutations/mutation_benchmarking.py [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+            if class_size not in artifacts:
+                artifacts[class_size] = {}
+            for mutation_name, mutations in mutation_sets.items():
+                artifacts[class_size][mutation_name] = SimulationSet(
+                    settings=SimulationSettings(
+                        scenario=scenario,
+                        student_provider=RealisticMockStudentProvider(class_size),
+                        cache_key=f"mutations/mutation_benchmarking/{mutation_name}/class_size_{class_size}/",
+                        initial_teams_provider=RealisticMockInitialTeamsProvider(
+                            class_size // team_size
```
Comment (Opeyem1a) [inline benchmarking/runs/mutations/mutation_benchmarking.py:143]: another thought on floor division - not sure if this makes sense when the numbers don't divide nicely, no need to change anything but take a double glance at it

File: benchmarking/runs/mutations/mutation_benchmarking_reasonable.py
Comment (Opeyem1a) [inline benchmarking/runs/mutations/mutation_benchmarking_reasonable.py:1]: if this is just the other run but improved, does the other one need to exist?

**Review by Opeyem1a — APPROVED** [review]
Generally fine, but left comments on things to peek at again

**Pull request merged**